test/integration/certificates/condition_list_type_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificates
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/jetstack/cert-manager/pkg/util"
+	"github.com/jetstack/cert-manager/test/integration/framework"
+)
+
+// Test_ConditionsListType ensures that the Certificate's Conditions API field
+// has been labelled as a list map. This is so that district field managers are
+// able to add and modify different Conditions without disrupting each others
+// entries. Conditions are keyed by `Type`.
+func Test_ConditionsListType(t *testing.T) {
```

Comment:
Suggestion: would it also make sense to add a test case where a condition is modified via 'Update' call (i.e from cmctl renew) and then modified again via Patch?

Reply:
> These tests are trying to assert that Certificate Conditions are correctly separated and indexed by the Type field on the condition, rather than testing Update vs Patch behaviour

Thanks for the comment. I thought that there could be a difference in behaviour when the condition is updated vs when it's patched- however this test does not look at it at that level.